proxy/handler_one2many.go

```diff
@@ -123,9 +123,28 @@ func (s *handler) forwardClientsToServerMultiUnary(sources []backendConnection,
 					return nil
 				}
 
+				// Send the header metadata first
+				md, err := src.clientStream.Header()
```

Comment:
from the [docs](https://pkg.go.dev/google.golang.org/grpc#ClientStream):

> If the metadata  is nil and the error is also nil, then the stream was terminated without headers, and the status can be discovered by calling RecvMsg.

So I would probably look into not calling `SetHeader()` if the `md` is nil below (and same for another function)

Reply:
ah, good point, done.